### PR TITLE
Update Maven Toolchain to Deploy Application 

### LINKED ISSUE #118: Build front-end with back-end

### LINKED ISSUE #119: Fix Spring-Cloud Dependency Version Problems
Spring boot server fails to load due to reported missing method.

Bug replicable by:

1. Going to top-level folder and running <code>mvn clean package</code>
2. Running the server with <code>java -jar target/dashboard[-version].jar</code>

logs:

image

### LINKED ISSUE #120: Successfully Deploy Application

### PR DESCRIPTION
**Pull request by samuelsogbesan**
This PR integrates changes to the pom.xml file that enables us to deploy our application.

- **Add frontend-maven-plugin to build front-end** - Creates a copy of [node.js](https://nodejs.org/en/) and temporarily installs frontend dependencies to run our production build script, <code>npm run build</code>.
[frontend-maven-plugin is distributed under the Apache 2.0 License](https://github.com/eirslett/frontend-maven-plugin/blob/master/LICENSE)
- **Add [maven-resources-plugin](https://maven.apache.org/plugins/maven-resources-plugin/) to move front-end build files to target/static so they can be served from backend**
maven-resource-plugin is distributed by Apache Maven.

- **Fix spring-cloud dependencies** - Use the [spring-cloud Bill of Materials](https://spring.io/projects/spring-cloud-gcp) to manage versions for our spring-cloud dependencies, including the spring-cloud datastore integration.

_Changes were made to the **.gitignore** to ignore the temporary version of node that is created in our front-end is built. This is made to provide maven with a copy of node so that it can access the dependencies and run our [parcel build script](https://parceljs.org/getting_started.html)_

A clean build of the project can be creating by navigating to the top-level folder and using <code>mvn clean package</code>

More information on deployment will be available on the [wiki page](https://github.com/googleinterns/step240-2020/wiki/Deployment-(Running-the-Server)). 
To run the server, navigate to the top-level folder and run <code>java -jar target/dashboard-0.0.1-SNAPSHOT.jar</code>

**PLEASE NOTE** to get this build to work, a problematic test file _src/test/java/com/google/graphgeckos/dashboard/storage/DatastoreRepositoryTests.java_ had to be commented out. This change will not be committed here, but it is expected these tests will be updated to work in a future PR.

Fixes #119 
Closes #118 
Closes #120

EDIT: Add Licensing Information.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -117,6 +121,72 @@
...
+            <id>Install node and npm</id>
+            <goals>
+              <goal>install-node-and-npm</goal>
+            </goals>
+            <phase>generate-resources</phase>
+            <configuration>
+              <nodeVersion>v10.14.2</nodeVersion>
+              <npmVersion>6.14.8</npmVersion>
+            </configuration>
+          </execution>
```
Comment (MagoDelBlocco) [inline pom.xml:140]: nit: add a blank line between each logical block, for readability purposes

File: src/test/java/com/google/graphgeckos/dashboard/storage/DatastoreRepositoryTests.java [hunk 2]
```diff
@@ -209,4 +209,4 @@ public void testGetLastRevisionsQueryAfterDeletion() throws IOException, Interru
     Assert.assertEquals(results.get(0), dummy2);
     Assert.assertEquals(results.get(1), dummy1);
   }
-}
+}
```
Comment (MagoDelBlocco) [inline src/test/java/com/google/graphgeckos/dashboard/storage/DatastoreRepositoryTests.java]: [add new line to the end of file](https://stackoverflow.com/questions/2287967/why-is-it-recommended-to-have-empty-line-in-the-end-of-a-source-file#:~:text=The%20empty%20line%20in%20the,you%20have%20reached%20the%20end.)

**Review by MagoDelBlocco — APPROVED** [review]
LGTM

File: src/test/java/com/google/graphgeckos/dashboard/storage/DatastoreRepositoryTests.java (hunk 2 above)
Comment (samuelsogbesan) [reply src/test/java/com/google/graphgeckos/dashboard/storage/DatastoreRepositoryTests.java]: File reverted to master. line space should be preserved.

File: pom.xml (hunk 1 above)
Comment (samuelsogbesan) [reply pom.xml:140]: Blank lines added between <code>execution</code> blocks with commit c980571

File: pom.xml [hunk 3]
```diff
@@ -117,6 +121,74 @@
...
+          <workingDirectory>src/main/webapp</workingDirectory>
+        </configuration>
+      </plugin>
+
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-resources-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>Copy my react app into my Spring Boot target static folder</id>
```
Comment (krasimir-google) [inline pom.xml:175]:
```suggestion
            <id>Copy React app into Spring Boot target static folder</id>
```

**Review by krasimir-google — APPROVED** [review]
Thank you!

Left one nit.

Please make sure this is rebased with the master branch when merging.

**Pull request merged**